### DISCUSSION TITLE
Transpiling to Haxe?

### DISCUSSION THREAD

**Discussion by ghost** [pfusik/cito#119] [General]
Haxe is more powerful than Fusion, and I think it's nothing wrong to leverage its abilities.

**Comment by pfusik** [comment]
What do you mean by "more powerful" ?
I haven't used Haxe, but targeting it is probably feasible.

**Reply by ghost** [reply]
Read about it on wiki and you will quickly realize that it's much more powerful than Fusion.

https://en.wikipedia.org/wiki/Haxe

**Comment by pfusik** [comment]
I've read about Haxe in 2000s, before I started my work on Fusion. They have different design goals: in Haxe you create whole apps, in Fusion you create components to be used from other languages. Haxe has syntax similar to the (now dead) ActionScript, Fusion is similar to C#. Fusion transpiles to C, D, Swift, TypeScript, OpenCL and Haxe does not.

There's a request to compare the two languages: #94

**Reply by ghost** [reply]
What do you think about transpiling to Haxe?

**Reply by pfusik** [reply]
I think it's a good idea, but PHP, Ruby and Go have priority as they are more popular.

**Comment by caesay** [comment]
I think this is an important conversation. Haxe has a lot of capabilities in it's standard lib, stuff like json/xml parsing, HTTP client, starting processes, etc. As you mentioned, it doesn't really produce a "lib" to be used from other languages, it's design goal is to produce a whole compiled application. 

For those of us looking to create re-usable libraries targeting multiple languages, HAXE is not suitable, and Fusion is designed for this purpose, but it's missing a lot of crucial std features which makes writing a library of any significance nearly impossible. Without process/start capabilities, I can't even offload the heavy lifting to another process and just provide a Fusion wrapper. It would be great if we could find a way to leverage HAXE's std lib in some useful way to expand the capabilities of Fusion

**Reply by pfusik** [reply]
I expect Fusion's library to grow with time. It is currently very low-level, but so are the C and C++ standard libraries. Let's continue this topic in #139.

Transpiling to Haxe is #121 and I think it is of less importance. Haxe is much more popular than Fusion, but far from being a mainstream language.

**Reply by caesay** [reply]
I agree, transpiling to HAXE is probably not important, but I was initially wondering if we could somehow leverage parts of their std lib. Having now looked into it, this will not be easy, because a lot of their std lib comes pre-compiled as binaries instead of source code.
